File: Lodjinha/app/src/main/java/br/com/bsavoini/lodjinha/adapters/BannerPagerAdapter.java
package br.com.bsavoini.lodjinha.adapters;

import android.support.v4.view.PagerAdapter;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import br.com.bsavoini.lodjinha.R;
import br.com.bsavoini.lodjinha.api.model.BannerModel;
import br.com.bsavoini.lodjinha.callbacks.BannerClickCallback;
import com.squareup.picasso.Picasso;

import java.util.List;

public class BannerPagerAdapter extends PagerAdapter {
    private List<BannerModel> banners;
    private BannerClickCallback bannerClickCallback;

    public BannerPagerAdapter(List<BannerModel> banners, BannerClickCallback bannerClickCallback) {
        this.banners = banners;
        this.bannerClickCallback = bannerClickCallback;
    }

    @Override
    public int getCount() {
        return banners.size();
    }

    @Override
    public boolean isViewFromObject(View view, Object obj) {
        return obj == view;
    }

    @Override
    public Object instantiateItem(ViewGroup container, int position) {
        LayoutInflater inflater = LayoutInflater.from(container.getContext());
        View view = inflater.inflate(R.layout.item_banner, container, false);
        container.addView(view);

        ImageView imgBanner = view.findViewById(R.id.img_banner);

        final BannerModel bannerModel = banners.get(position);
        Picasso.with(container.getContext())
                .load(bannerModel.getImageURL())
                .placeholder(R.drawable.menu_pattern)
                .into(imgBanner);

        view.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                bannerClickCallback.onClickBanner(bannerModel.getLinkURL());
            }
        });

        return view;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
        container.removeView((View) object);
    }

}

File: app/src/main/java/com/eric/alodjinha/features/home/adapter/BannerPagerAdapter.kt
package com.eric.alodjinha.features.home.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.FrameLayout
import androidx.viewpager.widget.PagerAdapter
import com.bumptech.glide.Glide
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions
import com.eric.alodjinha.R
import com.eric.alodjinha.features.home.model.Banner
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.frame_layout_banner.view.*

class BannerPagerAdapter(var mContext: Context, internal var banners: List<Banner>?) : PagerAdapter() {

    var mLayoutInflater: LayoutInflater = mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater

    override fun getCount(): Int {
        return banners!!.size
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {
        return view === `object` as FrameLayout
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val itemView = mLayoutInflater.inflate(R.layout.frame_layout_banner, container, false)

        val banner = banners!![position]

        Picasso.get()
            .load(banner.urlImagem)
            .placeholder(R.drawable.shopping_bag)
            .error(R.drawable.warning)
            .into(itemView.imageViewBanner)

        container.addView(itemView)

        return itemView
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        container.removeView(`object` as FrameLayout)
    }
}